### PR TITLE
Use single quotes in ember core and enforce it

### PR DESCRIPTION
**Pull request by jayphelps**
brace yourself

Since we're nearing 2.0, figured it would be time to start having a consistent quote usage. :fire: flame on.

### CODE REVIEW

File: .jscsrc [hunk 1]
```diff
@@ -33,6 +33,7 @@
   "disallowTrailingWhitespace": true,
   "disallowTrailingComma": true,
   "requireSpaceBeforeBlockStatements": true,
+  "validateQuoteMarks": "'",
```
Comment (jayphelps) [inline .jscsrc]: This enforces it at the jsc level btw

**Comment by jayphelps** [comment]
I don't know what's up with Travis. Does it just need to be kicked as usual or is there something I'm missing? (Read logs, don't see reason)

**Comment by rwjblue** [comment]
I restarted cross browser tests.

**Comment by rwjblue** [comment]
I have previously advocated for single quote (in JS not HBS/HTML) enforcement in Ember.

When I brought it up (just a couple months ago) with @wycats he mentioned that the code base at one point had different meaning for single quote and double quote (one is 'stuff dynamic from the user' and the other is 'static all the time'). This was extremely surprising to me (I had never heard it before and have certainly never followed...). 

---

TLDR; I am :+1:, but since he was previously opposed we will need :+1: / :-1: from @wycats before landing....

**Comment by jayphelps** [comment]
Given the nature of the PR, it's obviously going to keep getting conflicts as others commits are merged before it. Instead of playing cat and mouse, I'm going to wait until someone gives me a :+1: for merge, then fix the final conflicts.

**Comment by rwjblue** [comment]
@jayphelps - Understood (and I agree).  I'll try to make sure to bring it up again at one of the upcoming meetings....

**Comment by jayphelps** [comment]
@rwjblue All green.

**Comment by cibernox** [comment]
rebase

**Comment by jayphelps** [comment]
@cibernox sorry friend...it just had to done. :tada:
